bitnami/kubernetes-event-exporter/templates/deployment.yaml

```
@@ -72,20 +102,40 @@ spec:
                 name: {{ include "common.tplvalues.render" (dict "value" .Values.extraEnvVarsSecret "context" $) }}
             {{- end }}
           {{- end }}
-          resources: {{- toYaml .Values.resources | nindent 12 }}
-          {{- if .Values.containerSecurityContext.enabled }}
-          # yamllint disable rule:indentation
-          securityContext: {{- omit .Values.containerSecurityContext "enabled" | toYaml | nindent 12 }}
-          # yamllint enable rule:indentation
+          {{- if not .Values.diagnosticMode.enabled }}
+          {{- if .Values.lifecycleHooks }}
+          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.lifecycleHooks "context" $) | nindent 12 }}
+          {{- end }}
```

Why would you exclude the hook when enabling diagnostic mode?
```suggestion
          {{- if .Values.lifecycleHooks }}
          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.lifecycleHooks "context" $) | nindent 12 }}
          {{- end }}
          {{- if not .Values.diagnosticMode.enabled }}
```

---

Not sure if it is intentional or a typo, but I have recently seen that other charts disable `lifecycleHooks` when `diagnosticMode` is enabled. Those charts are Redis, Oauth2-proxy, Postgresql-ha, and Contour.

---

I'd say it's a typo